packages/api/src/auth/decoders/supabase.ts

```diff
@@ -0,0 +1,20 @@
+import jwt from 'jsonwebtoken'
+
+export const supabase = (token: string) => {
+  if (process.env.NODE_ENV === 'production') {
+    if (!process.env.SUPABASE_JWT_SECRET) {
+      throw new Error('`SUPABASE_JWT_SECRET` env var is not set.')
+    }
+
+    try {
+      const secret = process.env.SUPABASE_JWT_SECRET as string
+      return Promise.resolve(
```

Comment:
Sorry, I do not understand why do these return Promises -- when the Netlify decoded that also uses jwt does not have to?

https://github.com/redwoodjs/redwood/blob/1185f0f0b06c11004e7c0d86c19d0cd0c37b3c34/packages/api/src/auth/decoders/netlify.ts#L17

I see that the Auth0 decoder does

https://github.com/redwoodjs/redwood/blob/1185f0f0b06c11004e7c0d86c19d0cd0c37b3c34/packages/api/src/auth/decoders/auth0.ts#L28

but I thought that was because it had an API call with a callback to verify the key?

Reply:
I needed to return a rejected promise if the token verification fails. And if one return path returns a promise I want all of them to do that to be consistent.

Also, this method is invoked as the return value of `decodeToken` in `index.ts`, and the return value of that function is always a promise, so even if I did just do `return jwt.decode(token)`, it would still be implicitly wrapped in a promise there.